servers/zms/src/test/java/com/yahoo/athenz/zms/ZMSImplTest.java

```diff
@@ -25193,40 +25193,22 @@ public void testRoleTagsLimit() throws Exception {
         List<String> tagValues = Arrays.asList("val1", "val2", "val3", "val4");
         Role role = createRoleObject(domainName, roleName, null);
         role.setTags(Collections.singletonMap(tagKey, new StringList().setList(tagValues)));
-        zmsTest.putRole(mockDomRsrcCtx, domainName, roleName, auditRef, role);
-        Role dbRole = zmsTest.getRole(mockDomRsrcCtx, domainName, roleName, false, false, false);
-        
-        // expect for only 3 first tags to be presented 
-        assertEquals(dbRole.getTags().get(tagKey).getList().size(), 3);
-        assertTrue(dbRole.getTags().get(tagKey).getList().containsAll(
-            Arrays.asList("val1", "val2", "val3")));
-
-        // new tags map - 2 key with 2 values
-        Map<String, StringList> tags = new HashMap<>();
-        tags.put("newKey", new StringList().setList(Arrays.asList("newVal1", "newVal2")));
-        tags.put("newKey2", new StringList().setList(Arrays.asList("newVal3", "newVal4")));
-        RoleMeta rm = new RoleMeta().setTags(tags);
+        try {
+            zmsTest.putRole(mockDomRsrcCtx, domainName, roleName, auditRef, role);
+            fail();
+        } catch(ResourceException ex) {
+            assertEquals(ex.getCode(), ResourceException.BAD_REQUEST);
+            assertTrue(ex.getMessage().contains("role tag quota exceeded - limit: 3, current tags count: 0, new tags count: 4"));
+        }
 
-        // update role tags using role meta
-        zmsTest.putRoleMeta(mockDomRsrcCtx, domainName, roleName, auditRef, rm);
-        dbRole = zmsTest.getRole(mockDomRsrcCtx, domainName, roleName, false, false, false);
-        
-        // processing order of Map is not guarantee 
-        // expect for 2 tags from one key, and only one from the other..
-        if (dbRole.getTags().get("newKey").getList().size() == 2) {
-            assertTrue(dbRole.getTags().get("newKey").getList().containsAll(
-                Arrays.asList("newVal1", "newVal2")));
-
-            assertEquals(dbRole.getTags().get("newKey2").getList().size(), 1);
-            assertTrue(dbRole.getTags().get("newKey2").getList().contains("newVal3"));
-        } else {
-            assertEquals(dbRole.getTags().get("newKey").getList().size(), 1);
-            assertTrue(dbRole.getTags().get("newKey").getList().contains("newVal1"));
-            
-            assertEquals(dbRole.getTags().get("newKey2").getList().size(), 2);
-            assertTrue(dbRole.getTags().get("newKey2").getList().containsAll(
-                Arrays.asList("newVal3", "newVal4")));
+        try {
+            // role should not be created if fails to process tags..
+            Role dbRole = zmsTest.getRole(mockDomRsrcCtx, domainName, roleName, false, false, false);
```

Comment:
No need to have the "Role dbRole =" part since we expect the call throw an exception. this is usually tagged as unused variable in code analyzers.